### PR TITLE
Raise XPU tolerances for bf16 ResNet & BotNet TorchBench

### LINKED ISSUE #1305: Models got fail accuracy on BMG but passed on PVC
### 🐛 Describe the bug

```
python benchmarks/dynamo/timm_models.py --accuracy --float16 -d xpu -n10 --training --only botnet26t_256 --backend=inductor
```

Category | Model | Accuracy
-- | -- | --
timm_models_float16_training | botnet26t_256 | fail_accuracy
torchbench_float32_inference | detectron2_maskrcnn_r_50_fpn | fail_accuracy
timm_models_amp_fp16_training | fbnetv3_b | fail_accuracy
timm_models_bfloat16_training | fbnetv3_b | fail_accuracy
timm_models_amp_bf16_training | gernet_l | fail_accuracy
torchbench_float16_training | hf_Longformer | fail_accuracy
torchbench_bfloat16_training | hf_Longformer | fail_accuracy

### Versions

Environments:
Name| Version
-- | --
Device | BMG
OS | Ubuntu 24.10
Driver | 24.45.31740
Torch | v2.6.0-rc5
Torch-xpu-ops | release/2.6
Triton | 3.2.0
Transformers | 243e186efbf7fb93328dd6b34927a4e8c8f24395
Torchvision | 0.21.0+xpu
Torchaudio | 2.6.0+xpu
Torchbench | 766a5e3a189384659fd35a68c3b17b88c761aaac
Timms | ac3470188b914c5d7a5058a7e28b9eb685a62427
Bundle | 2025.0.4.20241205

### LINKED ISSUE #1799: Torchbench phlippe_resnet bfloat16 training got fail_accuracy
### 🐛 Describe the bug

phlippe_resnet  accuracy failed on both LTS and Rolling driver in 07/02 nightly tests
https://github.com/intel/torch-xpu-ops/actions/runs/16001019449
```
loading model: 0it [00:00, ?it/s]
loading model: 0it [00:11, ?it/s]
xpu  train phlippe_resnet                     
E0701 15:17:51.502000 2472533 site-packages/torch/_dynamo/utils.py:3007] RMSE (res-fp64): 0.00734, (ref-fp64): 0.00047 and shape=torch.Size([]). res.dtype: torch.bfloat16, multiplier: 3.000000, tol: 0.001000, use_larger_multiplier_for_smaller_tensor: 0
fail_accuracy

```

Reproducer
```
# Get pytorch wheel from https://github.com/intel/torch-xpu-ops/actions/runs/16001019449
pip install /path/to/pytorch/wheel
git clone https://github.com/pytorch/benchmark
cd benchmark
pip install -r requirements.txt
python install.py phlippe_resnet 
cd ..
git clone https://github.com/pytorch/pytorch
cd pytorch
git checkout 1586521461c8dc642735466fc143b7d366a858d0
python benchmarks/dynamo/torchbench.py --accuracy --bfloat16 -d xpu \
     -n10 --training --only phlippe_resnet --cold-start-latency --backend=inductor
```

### Versions

```
Device: PVC 1100
Driver: Rolling 25.13.33276
IGC: 2.10.11-1133~22.04
L0: 1.21.1.0-1123~22.04
Pytorch: 1586521461c8dc642735466fc143b7d366a858d0
Torch-xpu-ops: 9bf5973052feb696bf5776a549f5958cb64f7e55
```

### PR DESCRIPTION
**Pull request by astachowiczhabana**
Multiple TorchBench models on XPU fail accuracy tests due to numeric tolerance being too strict rather. Two contributing factors identified:

1. Measurement methodology change (PyTorch 2.6.0 enforcing cosine_similarity https://github.com/pytorch/pytorch/blob/main/benchmarks/dynamo/common.py#L2227) surfaced limitations and increased sensitivity in error checks for phlippe_resnet.
2. BatchNorm decomposition noise (~1e-5 RMSE per BN in fp16) accumulates through the iteration in botnet26t_256, pushing aggregate diffs beyond current thresholds.

**Analysis**

- phlippe_resnet failures reproduce across CPU and XPU; fp16 already uses higher tolerance, implying bf16 thresholds are misaligned.
- Disabling BN decomposition brings botnet26t_256 outputs within tolerance; with decomposition enabled, cumulative numeric error is expected.
- CI health indicates changes are non-disruptive; failures, where present, are unrelated to these PRs.

Fixes https://github.com/intel/torch-xpu-ops/issues/1799
Fixes https://github.com/intel/torch-xpu-ops/issues/1305

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo @chenyang78

### CODE REVIEW

**Review by EikanWang — APPROVED** [review]

**Review by desertfire — APPROVED** [review]

File: benchmarks/dynamo/timm_models.py [hunk 1]
```diff
@@ -71,6 +71,10 @@ def pip_install(package):
     "mobilenetv3_large_100",
 }
 
+REQUIRE_HIGHER_TOLERANCE_FP16_XPU = {
+    "botnet26t_256",
```
Comment (desertfire) [inline benchmarks/dynamo/timm_models.py:75]: Not specific to this PR, but we should use .yaml file for timm_models as well.

Comment (EikanWang) [reply benchmarks/dynamo/timm_models.py:75]: Sounds good. I will take this AR.
